larpandora/LArPandoraInterface/LArPandoraGeometry.cxx

```diff
@@ -29,57 +29,76 @@ void LArPandoraGeometry::LoadDetectorGaps(LArDetectorGapList &listOfGaps)
+            const float deltaY(std::fabs(driftVolume1.GetCenterY() - driftVolume2.GetCenterY()));
+            const float deltaZ(std::fabs(driftVolume1.GetCenterZ() - driftVolume2.GetCenterZ()));
+            
             const float widthX(0.5f * (driftVolume1.GetWidthX() + driftVolume2.GetWidthX()));
+            const float widthY(0.5f * (driftVolume1.GetWidthY() + driftVolume2.GetWidthY()));
+            const float widthZ(0.5f * (driftVolume1.GetWidthZ() + driftVolume2.GetWidthZ()));
+            
             const float gapX(deltaX - widthX);
-
-            if (gapX < 0.f || gapX > maxDisplacement || deltaY > maxDisplacement || deltaZ > maxDisplacement)
+            const float gapY(deltaY - widthY);
+            const float gapZ(deltaZ - widthZ);
+            
+            if (!isDualPhase && (gapX < 0.f || gapX > maxDisplacement || deltaY > maxDisplacement || deltaZ > maxDisplacement))
                 continue;
-
+            
             const float X1((driftVolume1.GetCenterX() < driftVolume2.GetCenterX()) ? (driftVolume1.GetCenterX() + 0.5f * driftVolume1.GetWidthX()) :
-                (driftVolume2.GetCenterX() + 0.5f * driftVolume2.GetWidthX()));
+                           (driftVolume2.GetCenterX() + 0.5f * driftVolume2.GetWidthX()));
             const float X2((driftVolume1.GetCenterX() > driftVolume2.GetCenterX()) ? (driftVolume1.GetCenterX() - 0.5f * driftVolume1.GetWidthX()) :
-                (driftVolume2.GetCenterX() - 0.5f * driftVolume2.GetWidthX()));
+                           (driftVolume2.GetCenterX() - 0.5f * driftVolume2.GetWidthX()));
             const float Y1(std::min((driftVolume1.GetCenterY() - 0.5f * driftVolume1.GetWidthY()),
-                (driftVolume2.GetCenterY() - 0.5f * driftVolume2.GetWidthY())));
+                                    (driftVolume2.GetCenterY() - 0.5f * driftVolume2.GetWidthY())));
             const float Y2(std::max((driftVolume1.GetCenterY() + 0.5f * driftVolume1.GetWidthY()),
-                (driftVolume2.GetCenterY() + 0.5f * driftVolume2.GetWidthY())));
+                                    (driftVolume2.GetCenterY() + 0.5f * driftVolume2.GetWidthY())));
             const float Z1(std::min((driftVolume1.GetCenterZ() - 0.5f * driftVolume1.GetWidthZ()),
-                (driftVolume2.GetCenterZ() - 0.5f * driftVolume2.GetWidthZ())));
+                                    (driftVolume2.GetCenterZ() - 0.5f * driftVolume2.GetWidthZ())));
             const float Z2(std::max((driftVolume1.GetCenterZ() + 0.5f * driftVolume1.GetWidthZ()),
-                (driftVolume2.GetCenterZ() + 0.5f * driftVolume2.GetWidthZ())));
+                                    (driftVolume2.GetCenterZ() + 0.5f * driftVolume2.GetWidthZ())));
+            
+            if (isDualPhase && (std::fabs(gapY) > maxDisplacement || std::fabs(gapZ) > maxDisplacement))
+            {
+                listOfGaps.push_back(LArDetectorGap(X1, Y1 + widthY, Z1 + widthY, X2, Y2 - widthY, Z2 - widthZ));
```

Comment:
Currently discussing this line with Etienne offline.

Reply:
Following discussion with Etienne it's been established that the average volume widths defined by widthY and widthZ are equivalent to picking the width of any given volume. Given that this is the case, in the future it may be worth reviewing this definition for widthY/Z and simplifying it to use the width of the volume in question directly, to improve clarity. I don't view this as high priority.